nodejs12.x/cookiecutter-typescript-app-template/{{cookiecutter.project_name}}/template.yml

```diff
@@ -0,0 +1,93 @@
+# This is the SAM template that represents the architecture of your serverless application
+# https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-template-basics.html
+
+# The AWSTemplateFormatVersion identifies the capabilities of the template
+# https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/format-version-structure.html
+AWSTemplateFormatVersion: 2010-09-09
+Description: >-
+  {{cookiecutter.project_name}}
+
+# Transform section specifies one or more macros that AWS CloudFormation uses to process your template
+# https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/transform-section-structure.html
+Transform:
+- AWS::Serverless-2016-10-31
+
+# Resources declares the AWS resources that you want to include in the stack
+# https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resources-section-structure.html
+Resources:
+  # Each Lambda function is defined by properties:
+  # https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction
+
+  # This is a Lambda function config associated with the source code: get-by-id.js
+  getByIdFunction:
+    Type: AWS::Serverless::Function
+    Metadata:
+      BuildMethod: makefile
```

Comment:
What is the expected user experience for Windows users? This seems to imply that those users either need to always build in container. I don't have a windows machine to test with, but is there a compatible Windows make executable available? Not seeing anything obvious for how this should work in the SAM CLI docs

Reply:
> What is the expected user experience for Windows users? This seems to imply that those users either need to always build in container. I don't have a windows machine to test with, but is there a compatible Windows make executable available? Not seeing anything obvious for how this should work in the SAM CLI docs

I did this based off of the templates that existed for other node cookiecutter project templates in the SAM CLI. I believe the `BuildMethod: makefile` is metadata used just by the SAM teams CodeBuild process and not something that is user facing

Reply:
Thanks for picking this up and moving forward with it on your side! I've just not been able to get back to this and address the items in my original PR. It's been on my todo list buuuuuuut data center migration has taken all of my time :(

Reply:
@rli choco install make should work for this.

Reply:
> @rli choco install make should work for this.

Should the [Requirements](https://github.com/aws/aws-sam-cli-app-templates/pull/98/files#diff-90a747e9da686c7110278c51fd0f75947a127dd24df951cee5397533acad824cR5) section in the readme mention `make` ? Probably also should mention `choco install make`, else Windows users will tend to find http://gnuwin32.sourceforge.net/packages/make.htm via a web search, which is not a great experience.

Reply:
@sriram-mv what if we drop `make` completely and use a `postinstall` script instead?

Reply:
I'm unsure what you mean here, the build method still relies on `make`, could you be more explicit?

Reply:
The proposal is to drop the `makefile` build method and to treat it as if it were regular javascript. in this case, the `npm postinstall` script would perform the typescript compilation. A downside that I see is that the "develop" and "production build" workflows become conflated, which I'm not sure is desirable.

Reply:
Ah! I'd still like to see it within makefile in that case, because that would mean less number of steps to context switch out of sam cli. Is the concern that for windows users we are not sure if this works?

Reply:
The main concern is the strange requirement for `make` on windows, but otherwise works fine. `make` not being available in the form of an npm dependency is not the end of the world, however, and I feel like is a nicer develop/deploy experience overall.